txp/Pos/Txp/MemState/Class.hs

```diff
@@ -73,11 +76,15 @@ getMemPool = getTxpLocalData (STM.readTVar . txpMemPool)
 getTxpExtra :: (MonadIO m, MonadTxpMem e m) => m e
 getTxpExtra = getTxpLocalData (STM.readTVar . txpExtra)
 
+txpLocalDataLock :: MVar ()
+txpLocalDataLock = unsafePerformIO $ newMVar ()
```

Comment:
Is it really necessary to use global variable here? Can't it be added to reader?